ComponentKit/Core/Scope/CKComponentScopeRootFactory.h

```diff
@@ -0,0 +1,17 @@
+/*
+ *  Copyright (c) 2014-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under the BSD-style license found in the
+ *  LICENSE file in the root directory of this source tree. An additional grant
+ *  of patent rights can be found in the PATENTS file in the same directory.
+ *
+ */
+
+#import <Foundation/Foundation.h>
+
+@protocol CKComponentStateListener;
+
+@class CKComponentScopeRoot;
+
+CKComponentScopeRoot *CKComponentScopeRootWithListener(id<CKComponentStateListener> listener);
```

Comment:
This is the normal factory that should be used with CKComponent implementations. For working ranges implementation we'd like to override the predicates for controllers to make the tree traversal more efficient, so may need a way to inject this somehow into the hosting view and datasource. Thoughts?